core/nix/src/internal/Tzfile.kt

```diff
@@ -0,0 +1,353 @@
+            reader.readInt(),
+            reader.readInt(),
+            reader.readInt(),
+            reader.readInt(),
+            reader.readInt()
+        ).also {
+            check(it.ttisutcnt == 0 || it.ttisutcnt == it.typecnt)
+            check(it.ttisstdcnt == 0 || it.ttisstdcnt == it.typecnt)
+        }
+    }
+
+    val header = readHeader()
+    return when (header.version) {
+        1 -> {
+            TzFile(reader.read32BitData(header), null)
+        }
+        else -> {
+            reader.read32BitData(header) // skipped
+            val newHeader = readHeader()
+            val parsedData = reader.read64BitData(newHeader)
+            val footer = reader.readFooter()
+            TzFile(parsedData, footer)
+        }
+    }
+}
+
+internal class PosixTzString(
+    private val standardTime: Pair<String, UtcOffset>,
+    private val daylightTime: Pair<String, UtcOffset>?,
+    private val rules: Pair<MonthDayTime, MonthDayTime>?,
+) {
+    companion object {
+        /**
+         * Reads a POSIX TZ string from the [reader] if it is present, or returns `null` if it is not.
+         *
+         * The string format is described in * https://pubs.opengroup.org/onlinepubs/9699919799/, section 8.3,
+         * with additional extensions in https://datatracker.ietf.org/doc/html/rfc8536#section-3.3.1
+         *
+         * @throws IllegalArgumentException if the string is invalid
+         * @throws IllegalStateException if the string is invalid
+         */
+        fun readIfPresent(reader: BinaryDataReader): PosixTzString? = reader.readPosixTzString()
+    }
+
+    fun toRecurringZoneRules(): RecurringZoneRules? {
+        /**
+         * In theory, it's possible to have a DST transition but no start/end date.
+         * In this case, the behavior is not specified, and on Linux, the rules for America/New_York are used
+         * to determine the start/end dates (see `tzset(3)`, search for `posixrules`).
+         * The <https://github.com/HowardHinnant/date> library takes the lack of start/end dates
+         * to mean that the standard offset is always in effect, which seems to be a much more reasonable interpretation.
+         */
+        if (daylightTime == null || rules == null) return null
+        val (start, end) = rules
+        val rule1 = RecurringZoneRules.Rule(start, standardTime.second, daylightTime.second)
+        val rule2 = RecurringZoneRules.Rule(end, daylightTime.second, standardTime.second)
+        return RecurringZoneRules(listOf(rule1, rule2))
+    }
+}
+private fun BinaryDataReader.readPosixTzString(): PosixTzString? {
+    var c = readAsciiChar()
+    fun readName(): String? {
+        if (c == '\n') return null
+        var name = ""
+        if (c == '<') {
+            c = readAsciiChar()
+            while (c != '>') {
+                check(c.isLetterOrDigit() || c == '-' || c == '+') { "Invalid char '$c' in the std name in POSIX TZ string" }
+                name += c
```

Comment:
How long is usually a name constructed this way?

Reply:
This is for abbreviations, so three or four ASCII chars.

Reply:
I'd suggest to use a StringBuilder then.

Reply:
Ok. Is this a matter of style? Because I didn't see any difference in performance.